cycles-ledger/tests/tests.rs

```diff
@@ -930,59 +947,917 @@ fn test_withdraw_fails() {
+            // other Ledgers because the operation transfers cycles.
+            effective_fee: Some(env.icrc1_fee()),
+            // The timestamp is set by the ledger.
+            timestamp: env.nanos_since_epoch_u64(),
+            transaction: Transaction {
+                // The created_at_time was set to now.
+                created_at_time: Some(now),
+                // The transaction.memo is the canister ID receiving the cycles
+                // encoded in cbor as object with a 'receiver' field marked as 0.
+                memo: Some(encode_withdraw_memo(&withdraw_receiver)),
+                // Withdrawals are recorded as burns.
+                operation: Operation::Burn {
+                    from: account1_3,
+                    spender: Some(withdrawer1),
+                    // The transaction.operation.amount is the withdrawn amount.
+                    amount: withdraw_amount,
+                },
+            },
+        },
+    );
+
+    // withdraw cycles using spender subaccount
+    let withdraw_receiver_balance = env.state_machine.cycle_balance(withdraw_receiver);
+    let withdraw_amount = 500_000_000_u128;
+    env.icrc2_approve_or_trap(
+        account1_4.owner,
+        ApproveArgs {
+            from_subaccount: account1_4.subaccount,
+            spender: withdrawer1_1,
+            amount: Nat::from(withdraw_amount + FEE),
+            expected_allowance: None,
+            expires_at: None,
+            fee: None,
+            memo: None,
+            created_at_time: None,
+        },
+    );
+    let withdraw_idx = env.withdraw_from_or_trap(
+        withdrawer1_1.owner,
+        WithdrawFromArgs {
+            from: account1_4,
+            to: withdraw_receiver,
+            created_at_time: None,
+            amount: Nat::from(withdraw_amount),
+            spender_subaccount: withdrawer1_1.subaccount,
+        },
+    );
+    assert_eq!(
+        withdraw_receiver_balance + withdraw_amount,
+        env.state_machine.cycle_balance(withdraw_receiver)
+    );
+    assert_eq!(
+        env.icrc2_allowance(account1_4, withdrawer1_1),
+        Allowance {
+            allowance: 0_u128.into(),
+            expires_at: None
+        }
+    );
+    assert_eq!(
+        env.icrc1_balance_of(account1_4),
+        1_000_000_000 - withdraw_amount - FEE - FEE
+    );
+    expected_total_supply -= withdraw_amount + FEE + FEE;
+    assert_eq!(env.icrc1_total_supply(), expected_total_supply);
+
+    // check that the burn block created is correct
+    assert_display_eq(
+        &env.get_block(withdraw_idx.clone()),
+        &Block {
+            // The new block parent hash is the hash of the last deposit.
+            phash: Some(env.get_block(withdraw_idx - 1u8).hash().unwrap()),
+            // The effective fee of a burn block created by a withdrawal
+            // is the fee of the Ledger. This is different from burn in
+            // other Ledgers because the operation transfers cycles.
```

Comment:
```suggestion
            // is the fee of the ledger. This is different from burn in
            // other ledgers because the operation transfers cycles.
```